File: src/005/p005.go
package main

import "fmt"

func gcd(a, b int) int {
	for b > 0 {
		mod := a % b
		a, b = b, mod
	}
	return a
}

func lcm(a, b int) int {
	return a * b / gcd(a, b)
}

func smallestDividedBy(limit int) int {
	result := 1
	for i := 2; i < limit; i++ {
		result = lcm(result, i)
	}
	return result
}

func solve() int {
	// Smallest number divisible by range 1 to 20.
	return smallestDividedBy(20)
}

func main() {
	result := solve()
	fmt.Println(result)
}

File: src/005/p005.lua
function gcd(a, b)
  while b > 0 do
    local mod = a % b
    a = b
    b = mod
  end
  return a
end

function lcm(a, b)
  return a * b / gcd(a, b)
end

function smallest_divided_by(limit)
  local result = 1
  for i=2, limit do
    result = lcm(result, i)
  end
  return math.floor(result)
end

function solve()
  -- Smallest number divisible by range 1 to 20.
  return smallest_divided_by(20)
end

local result = solve()
print(result)
